text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,502 @@
+* During CDK deploy:
+  * Assets are published to CDK Toolkit staging buckets in Management account
+  * Stack set create/update is executed in Management Account
+  * Stack set instance retrieves artifacts from management Account staging bucket
+
+### [4.1] Bootstrapping
+
+CDK apps that utilize assets require “Bootstrap” infrastructure for storing and publishing assets. This can either be done using the cdk bootstrap command which will deploy a default stack with all the required infrastructure or a customized template. 
+
+### Permissions
+Since stack set instances are deployed outside the CDK context and can be deployed to “external” accounts, child accounts need permissions to retrieving assets from the centralized staging bucket in the management account. This can be achieved by adding a bucket policy to the staging buckets:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Principal": "*",
+            "Action": "s3:GetObject",
+            "Resource": [
+                "arn:aws:s3:::cdk-hnb659fds-assets-00000000000/*"
+            ],
+            "Effect": "Allow",
+            "Condition": {
+                "StringEquals": {
+                    "aws:PrincipalOrgID": "o-123"
+                }
+            }
+        }
+    ]
+}
+```
+* External accounts can be added via principal array or if all the accounts are in the same organization, access can be enabled by organizational condition
+  * The caveat with using principal directly is the # of accounts is limited by bucket policy size
+* Since this is not a standard feature, users will need to deploy a customized bootstrap template
+
+Ref:
+* https://docs.aws.amazon.com/cdk/latest/guide/bootstrapping.html#bootstrapping-customizing
+
+### Multi-Region Bootstrapping
+When an S3 Asset such as Lambda code is referenced in CFN, the S3 asset must exist in the same region the stack is being deployed to. This means all regions a stack set deploys to, the region in the management account must also be bootstrapped. Users can either deploy these individually to each region, or create a stack set that deploys the customized template to each region.
+
+When creating the stack construct that is passed into the stack set construct described in section 2.2, the DefaultStackSynthesizer also needs to be modified with the file asset bucket name. This configures file asset references in the CFN template to reference non-aws account specific buckets. I.e. When stack set instances are deployed, they will reference buckets in the central account.
+
+```ts
+const stackDef = new cdk.Stack(app,
+  'stack-poc', 
+  {synthesizer: new DefaultStackSynthesizer(
+    {fileAssetsBucketName: 'cdk-${Qualifier}-assets-000000000000-${AWS::Region}'})});
+const stackSet = new cdk.StackSet(app, 'stack-set-poc',
+  {
+    stack: stackDef,
+    faultTolerancePercentage: 10,
+    maxConcurrentPercentage: 10,
+    regionConcurrencyType: 'PARALLEL'
+  }
+);
```

Comment:
Let's move this example to the "nice" API.

Reply:
This was just removed since it was part of option 2

Reply:
Still here, still not using the "nice" API (the indentation is also weird).

Reply:
Moved to "nice" API